esp/logging/loggingagent/espserverloggingagent/loggingagent.hpp

```diff
@@ -54,7 +54,12 @@ class CTransIDBuilder : public CInterface, implements IInterface
         if (value)
             id.append(value->get()).append('-');
         else
-            id.append('??-');
+        {
+            const char* ptr = key;
+            if (strlen(key) > 11) //skip 'Transaction'
```

Comment:
Also how can we be sure the key will always contain the prefix 'Transaction' ? at the the very least we might want to make a comment where ever the keys are defined alerting the fact that 'TransactionXYZ' is expected?